app/services/chart_builder.rb

```diff
@@ -1,15 +1,36 @@
 class ChartBuilder
-  def initialize(benchmark_runs)
+  attr_reader :data, :versions, :benchmark_result_type
+
+  def initialize(benchmark_runs = [], benchmark_result_type)
+    @benchmark_result_type = benchmark_result_type
     @benchmark_runs = benchmark_runs
+    # @versions is a hash that holds all version/environment information for each datapoint, 
+    #   and gets converted to html and stored in @data[:categories]
+    @versions = []
+    # @data[:columns] is JSON that looks like [{ name: "benchmark1", data: [1.1, 1.2] }]
+    # @data[:categories] is JSON that looks like [version_html_string, version_html_string]
```

Comment:
Might not matter with my other comments. I dont believe hash_to_html should be cached, but rather be put into the HTML directly